File: sources/net.sf.j2s.core/test/src/com/falstad/circuit/ImportExportClipboardDialog.js
(function(){var P$=Clazz.newPackage("com.falstad.circuit"),I$=[];
var C$=Clazz.newClass(P$, "ImportExportClipboardDialog", null, 'a2s.Dialog', ['com.falstad.circuit.ImportExportDialog', 'java.awt.event.ActionListener']);

C$.$clinit$ = function() {Clazz.load(C$, 1);
}

Clazz.newMeth(C$, '$init0$', function () {
var c;if((c = C$.superclazz) && (c = c.$init0$))c.apply(this);
this.cframe = null;
this.importButton = null;
this.closeButton = null;
this.text = null;
this.$type = null;
this.clipboard = null;
}, 1);

Clazz.newMeth(C$, '$init$', function () {
this.clipboard = null;
}, 1);

Clazz.newMeth(C$, 'c$$com_falstad_circuit_CirSim$com_falstad_circuit_ImportExportDialog_Action', function (f, type) {
C$.superclazz.c$$a2s_Frame$S$Z.apply(this, [f, (type === (I$[0]||(I$[0]=Clazz.load(Clazz.load('com.falstad.circuit.ImportExportDialog').Action))).EXPORT ) ? "Export" : "Import", false]);
C$.$init$.apply(this);
this.cframe = f;
this.setLayout$java_awt_LayoutManager(Clazz.new_((I$[1]||(I$[1]=Clazz.load('com.falstad.circuit.ImportExportDialogLayout')))));
this.add$java_awt_Component(this.text = Clazz.new_((I$[2]||(I$[2]=Clazz.load('a2s.TextArea'))).c$$S$I$I,["", 10, 60]));
this.importButton = Clazz.new_((I$[3]||(I$[3]=Clazz.load('a2s.Button'))).c$$S,["Import"]);
this.$type = type;
this.add$java_awt_Component(this.importButton);
this.importButton.addActionListener$java_awt_event_ActionListener(this);
this.add$java_awt_Component(this.closeButton = Clazz.new_((I$[3]||(I$[3]=Clazz.load('a2s.Button'))).c$$S,["Close"]));
this.closeButton.addActionListener$java_awt_event_ActionListener(this);
var x = this.cframe.main.getLocationOnScreen();
this.resize$I$I(400, 300);
var d = this.getSize();
this.setLocation$I$I(x.x + ((this.cframe.winSize.width - d.width)/2|0), x.y + ((this.cframe.winSize.height - d.height)/2|0));
}, 1);

Clazz.newMeth(C$, 'setDump$S', function (dump) {
this.text.setText$S(dump);
});

Clazz.newMeth(C$, 'execute', function () {
if (this.$type === (I$[0]||(I$[0]=Clazz.load(Clazz.load('com.falstad.circuit.ImportExportDialog').Action))).EXPORT ) this.text.selectAll();
this.setVisible$Z(true);
});

Clazz.newMeth(C$, 'actionPerformed$java_awt_event_ActionEvent', function (e) {
var i;
var src = e.getSource();
if (src === this.importButton ) {
{
this.cframe.readSetup$S(this.text.getText());
}}if (src === this.closeButton ) this.setVisible$Z(false);
});

Clazz.newMeth(C$, 'handleEvent$java_awt_Event', function (ev) {
if (ev.id == 201) {
this.cframe.main.requestFocus();
this.setVisible$Z(false);
this.cframe.impDialog = null;
return true;
}return C$.superclazz.prototype.handleEvent$java_awt_Event.apply(this, [ev]);
});

Clazz.newMeth(C$);
})();
//Created 2017-12-17 19:28:19

File: sources/net.sf.j2s.core/test/src/com/falstad/circuit/ImportExportClipboardDialog.java
package com.falstad.circuit;

import java.awt.Dimension;
import java.awt.Event;
import java.awt.Point;
import java.awt.event.*;
import java.awt.datatransfer.Clipboard;
import java.awt.datatransfer.DataFlavor;
import java.awt.datatransfer.StringSelection;
import java.awt.datatransfer.Transferable;

import a2s.Button;
import a2s.Dialog;
import a2s.TextArea;

class ImportExportClipboardDialog
extends Dialog
implements ImportExportDialog,ActionListener
{
    CirSim cframe;
    Button importButton, closeButton;
    TextArea text;
    Action type;

    Clipboard clipboard = null;

    ImportExportClipboardDialog(CirSim f, Action type) {
	super(f, (type == Action.EXPORT) ? "Export" : "Import", false);
	cframe = f;
	setLayout(new ImportExportDialogLayout());
	add(text = new TextArea("", 10, 60));
/*
	if (type == Action.EXPORT)
	    importButton = new Button("Copy to clipboard");
	else
*/
	    importButton = new Button("Import");
	this.type = type;
	add(importButton);
	importButton.addActionListener(this);
	add(closeButton = new Button("Close"));
	closeButton.addActionListener(this);
	Point x = cframe.main.getLocationOnScreen();
	resize(400, 300);
	Dimension d = getSize();
	setLocation(x.x + (cframe.winSize.width-d.width)/2,
		    x.y + (cframe.winSize.height-d.height)/2);
    }

    public void setDump(String dump)
    {
	text.setText(dump);
    }

    public void execute()
    {
	if ( type == Action.EXPORT )
	    text.selectAll();
	setVisible(true);
    }

    public void actionPerformed(ActionEvent e) {
	int i;
	Object src = e.getSource();
	if (src == importButton) {
/*
	    if (clipboard == null)
		clipboard = getToolkit().getSystemClipboard();
	    if ( type == Action.EXPORT )
	    {
		StringSelection data = new StringSelection(text.getText());
		clipboard.setContents(data, data);
	    }
	    else
*/
	    {
	        cframe.readSetup(text.getText());
	    }
	}
	if (src == closeButton)
	    setVisible(false);
    }

    public boolean handleEvent(Event ev) {
	if (ev.id == Event.WINDOW_DESTROY) {
	    cframe.main.requestFocus();
	    setVisible(false);
	    cframe.impDialog = null;
	    return true;
	}
	return super.handleEvent(ev);
    }

}
